stdlib/test/python/test_python_object.mojo

```diff
@@ -517,6 +517,94 @@ def test_setitem_raises():
         d[[1, 2, 3]] = 5
 
 
+fn test_py_slice() raises:
+    var a = PythonObject([1, 2, 3, 4, 5])
+    assert_equal("[2, 3]", str(a[1:3]))
+    assert_equal("[1, 2, 3, 4, 5]", str(a[:]))
+    assert_equal("[1, 2, 3]", str(a[:3]))
+    assert_equal("[3, 4, 5]", str(a[2:]))
+    assert_equal("[1, 3, 5]", str(a[::2]))
+    assert_equal("[2, 4]", str(a[1::2]))
+    assert_equal("[4, 5]", str(a[-2:]))
+    assert_equal("[1, 2, 3]", str(a[:-2]))
+    assert_equal("[5, 4, 3, 2, 1]", str(a[::-1]))
+    assert_equal("[1, 2, 3, 4, 5]", str(a[-10:10]))  # out of bounds
+    assert_equal("[1, 2, 3, 4, 5]", str(a[::]))
+    assert_equal("[1, 2, 3, 4, 5]", str(a[:100]))
+    assert_equal("[]", str(a[5:]))
+    assert_equal("[5, 4, 3, 2]", str(a[:-5:-1]))
+
+    var b = Python.evaluate("[i for i in range(1000)]")
+    assert_equal("[0, 250, 500, 750]", str(b[::250]))
+    with assert_raises(contains="slice step cannot be zero"):
+        _ = b[::0]
+    # Negative cases such as `b[1.3:10]` or `b["1":10]` are handled by parser
+    # which would normally throw a TypeError in Python
+
+    var s = PythonObject("Hello, World!")
+    assert_equal("Hello", str(s[:5]))
+    assert_equal("World!", str(s[7:]))
+    assert_equal("!dlroW ,olleH", str(s[::-1]))
+    assert_equal("Hello, World!", str(s[:]))
+    assert_equal("Hlo ol!", str(s[::2]))
+    assert_equal("Hlo ol!", str(s[None:None:2]))
+
+    var t = PythonObject((1, 2, 3, 4, 5))
+    assert_equal("(2, 3, 4)", str(t[1:4]))
+    assert_equal("(4, 3, 2)", str(t[3:0:-1]))
+
+    var empty = PythonObject([])
+    assert_equal("[]", str(empty[:]))
+    assert_equal("[]", str(empty[1:2:3]))
+
+    var d = Python.dict()
+    d["a"] = 1
+    d["b"] = 2
+    with assert_raises(contains="slice(1, 3, None)"):
```

Comment:
FYI this fails internally in CI since it says `unhashable type: 'slice'`.  I haven't looked closely as to why it fails internally but passes in OSS CI (perhaps different CPython versions as we're using 3.11 internally it looks like primarily).  I disabled the test so I can merge this, but we should look into it as a follow-up.